pkg/gateway/model_build_listener.go

```diff
@@ -28,33 +30,29 @@ func (t *latticeServiceModelBuildTask) extractListenerInfo(
 	if err != nil {
```

Comment:
Just to make the `extractListenerInfo()` more easy to follow (by reversing the `parentRef.SectionName != nil` condition and removing the `found := false` flag and returning eariler ).


 and override protocol string to "TLS_PASSTHROUGH" when create the lattice listener for k8s Gateway with `section.TLS.Mode==Passthrough`